thentos-core/src/Thentos/Backend/Core.hs

```diff
@@ -114,52 +117,58 @@ actionErrorToServantErr e = do
 
 
 class ThentosErrorToServantErr db where
-    thentosErrorToServantErr :: ThentosError db -> (Maybe (Priority, String), ServantErr)
+    -- | Convert a 'ThentosError' into a 'ServantErr'. If the first argument is 'Just' provided,
+    -- it must be invoked to create the returned error, otherwise a suitable default implementation
+    -- must be used for that purpose. This allows derived db's to adapt the details of the error
+    -- actually returned.
+    thentosErrorToServantErr :: Maybe MkServantErrFun -> ThentosError db
```

Comment:
Do we really need the extra `Maybe` argument here? It seems like if a derived db wanted to change the error, it could already do it (by applying whatever function [here](https://github.com/liqd/thentos/pull/282/files#diff-9520a252d8a4df0b5abfefea674c70f7R150)).